src/smt/theory_str_noodler/nielsen_decision_procedure.cpp

```diff
@@ -47,6 +47,13 @@ namespace smt::noodler {
     lbool NielsenDecisionProcedure::compute_next_solution() {
         // if the compute_next_solution was called for the first time
         if(this->graphs.size() == 0) {
+
+            // sort predicates from smallest to largest
+            std::vector<Predicate>& preds = this->formula.get_predicates();
+            std::sort(preds.begin(), preds.end(), [](const Predicate& p1, const Predicate& p2) {
+                return (p1.get_left_side().size() + p1.get_right_side().size()) < (p2.get_left_side().size() + p2.get_right_side().size());
```

Comment:
`get_left_side().size() + get_right_side().size()` might be a good candidate for its own method (it is also used in `get_formula_cost`)

Reply:
Good idea